### ISSUE TITLE
Oauth: Cannot sync role from github since migration to v10.1.1

### ISSUE THREAD

**Issue by arouene** [grafana/grafana#74688]
### What happened?

We used Grafana v9.5 and migrated to v10.1.1.

Grafana is configured to use auto-login with Github OAuth2.

Here is the configuration for authentication.

```
[auth]
disable_login_form = True
oauth_auto_login = True

[auth.github]
enabled = True
allow_sign_up = True
client_id = X
client_secret = X
scopes = user:email,read:org
auth_url = https://github.com/login/oauth/authorize
token_url = https://github.com/login/oauth/access_token
api_url = https://api.github.com/user
allowed_domains = 
team_ids = 
allowed_organizations = Org
role_attribute_path = contains(groups[*], '@Org/ops') && 'Admin' || contains(groups[*], '@Org/grafanaeditors') && 'Editor' || 'Viewer'
```

### What did you expect to happen?

We exepect to be able to log in and to have a role in relation to our github Teams, admin if in Ops, Editor if in GrafanaEditore, else Viewer.

We can log in just fine, but we are always Viewer

### Did this work before?

It we worked without changing anything in Grafana 9.5

In 9.5 we can see this in the logs in debug level : 

```
Sep 11 19:07:54 grafana grafana-server[1746890]: logger=oauth t=2023-09-11T19:07:54.748798312+02:00 level=debug msg="OAuthLogin got user info" userInfo="Id: 20049990, Name: Aurélien Rouëné, Email: aurelien.github.arouene@rouene.fr, Login: arouene, Role: Admin, Groups: [...edited...]"
```

That line does not appear in the logs of Grafana 10.1.1

### How do we reproduce it?

1. Use configuration above for authentication (have a github oauth2 configuration too)
2. Have users in github teams
3. Try to log-in with users in admin or editor teams
4. Check if they are admin or editor

### Is the bug inside a dashboard panel?

No

### Environment (with versions)?

Grafana: 10.1.1
OS: Linux (AlmaLinux 9.2)
Browser: Any

### Grafana platform?

A package manager (APT, YUM, BREW, etc.)

**Comment by mellieA** [comment]
@arouene What happens if you enable [debug logging](https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#log) for auth? The viewer role is default, so it could mean that the config isn't working. I also suggest to start with a single group to get it working, then add additional groups from there. 

https://grafana.com/docs/grafana/latest/setup-grafana/configure-security/configure-authentication/github/#map-roles-using-github-teams 

For general OSS help, we ask that you use our forum at community.grafana.com. I don't see reports from other users that they're having trouble with oauth2 in this release.

**Comment by arouene** [comment]
@mellieA, Yes the configuration isn't working, but it works just fine in 9.5.

I tried `role_attribute_path = 'Editor'` was not working either, but `role_attribute_path = "'Editor'"` is kind of working (Everyone is Editor).

**Comment by aarongodin** [comment]
Hi @arouene, I'm working on debugging this locally and am having trouble recreating the issue. I copied verbatim the config you posted, and substituted my own org for the `Org` name referenced in your values for `allowed_organizations` and `role_attribute_path`. The mapping seems to work correctly - I'm able to login to Grafana through GitHub and receive the correct Grafana role based on my GitHub team.

I'm trying to find any other aspect of config that could affect this. In the mean time, could you ensure a couple things:

1. Ensuring you have debug logging on as you described above, do you see a log entry that matches `logger=oauth.github url="https://api.github.com/user/teams?per_page=100"`. This is what indicates that grafana has fetched the teams for the user.
2. Are you using a real value for the `Org` string in your own config?

**Comment by neilharris123** [comment]
I am also experiencing this issue.
